patches/server/0431-Implement-Mob-Goal-API.patch

```diff
@@ -50,102 +50,14 @@ index 0000000000000000000000000000000000000000..dfc026d183adab1dde5942f36e7a281b
-+import org.bukkit.entity.Blaze;
-+import org.bukkit.entity.Cat;
-+import org.bukkit.entity.CaveSpider;
-+import org.bukkit.entity.ChestedHorse;
-+import org.bukkit.entity.Chicken;
-+import org.bukkit.entity.Cod;
-+import org.bukkit.entity.Cow;
-+import org.bukkit.entity.Creature;
-+import org.bukkit.entity.Creeper;
-+import org.bukkit.entity.Dolphin;
-+import org.bukkit.entity.Donkey;
-+import org.bukkit.entity.Drowned;
-+import org.bukkit.entity.ElderGuardian;
-+import org.bukkit.entity.EnderDragon;
-+import org.bukkit.entity.Enderman;
-+import org.bukkit.entity.Endermite;
-+import org.bukkit.entity.Evoker;
-+import org.bukkit.entity.Fish;
-+import org.bukkit.entity.Flying;
-+import org.bukkit.entity.Fox;
-+import org.bukkit.entity.Ghast;
-+import org.bukkit.entity.Giant;
-+import org.bukkit.entity.Golem;
-+import org.bukkit.entity.Guardian;
-+import org.bukkit.entity.Hoglin;
-+import org.bukkit.entity.Horse;
-+import org.bukkit.entity.Husk;
-+import org.bukkit.entity.Illager;
-+import org.bukkit.entity.Illusioner;
-+import org.bukkit.entity.IronGolem;
-+import org.bukkit.entity.Llama;
-+import org.bukkit.entity.MagmaCube;
-+import org.bukkit.entity.Mob;
-+import org.bukkit.entity.Monster;
-+import org.bukkit.entity.Mule;
-+import org.bukkit.entity.MushroomCow;
-+import org.bukkit.entity.Ocelot;
-+import org.bukkit.entity.Panda;
-+import org.bukkit.entity.Parrot;
-+import org.bukkit.entity.Phantom;
-+import org.bukkit.entity.Pig;
-+import org.bukkit.entity.PigZombie;
-+import org.bukkit.entity.Piglin;
-+import org.bukkit.entity.PiglinAbstract;
-+import org.bukkit.entity.PiglinBrute;
-+import org.bukkit.entity.Pillager;
-+import org.bukkit.entity.PolarBear;
-+import org.bukkit.entity.PufferFish;
-+import org.bukkit.entity.Rabbit;
-+import org.bukkit.entity.Raider;
-+import org.bukkit.entity.Ravager;
-+import org.bukkit.entity.Salmon;
-+import org.bukkit.entity.Sheep;
-+import org.bukkit.entity.Shulker;
-+import org.bukkit.entity.Silverfish;
-+import org.bukkit.entity.Skeleton;
-+import org.bukkit.entity.SkeletonHorse;
-+import org.bukkit.entity.Slime;
-+import org.bukkit.entity.Snowman;
-+import org.bukkit.entity.Spellcaster;
-+import org.bukkit.entity.Spider;
-+import org.bukkit.entity.Squid;
-+import org.bukkit.entity.Stray;
-+import org.bukkit.entity.Strider;
-+import org.bukkit.entity.Tameable;
-+import org.bukkit.entity.TraderLlama;
-+import org.bukkit.entity.TropicalFish;
-+import org.bukkit.entity.Turtle;
-+import org.bukkit.entity.Vex;
-+import org.bukkit.entity.Villager;
-+import org.bukkit.entity.Vindicator;
-+import org.bukkit.entity.WanderingTrader;
-+import org.bukkit.entity.WaterMob;
-+import org.bukkit.entity.Witch;
-+import org.bukkit.entity.Wither;
-+import org.bukkit.entity.WitherSkeleton;
-+import org.bukkit.entity.Wolf;
-+import org.bukkit.entity.Zoglin;
-+import org.bukkit.entity.Zombie;
-+import org.bukkit.entity.ZombieHorse;
-+import org.bukkit.entity.ZombieVillager;
++import org.bukkit.entity.*;
```

Comment:
no

Reply:
sorry I need to figure out these settings
anyway the goal key should not be for skeletons it should be for abstractskeletons

Reply:
Go to your IntelliJ settings, search for "Class count to" and find these two options under `Editor -> Code Style -> Java -> Imports`:
![image](https://user-images.githubusercontent.com/19861299/125917905-71274218-cca5-41e7-b9bb-ace541a40d2d.png)
Set the value to 999 or something stupidly high.

What does it actually fix, though? Can e.g. strays do melee attacks?

Reply:
sorry about the intelliJ stuff thank you
anyway, wither skeletons use the same melee goal as in abstractskeleton, but if it's considered to be a skeleton goal then we can't apply it to wither skeletons.